### ISSUE TITLE
Fastify does not support custom JSON Content-types

### ISSUE THREAD

**Issue by saulotoledo** [fastify/fastify#2307]
##  🐛 Bug Report

Fastify does not support custom JSON Content-types such as `application/ld+json`, `application/hal+json` and `application/vnd.api+json`, required by HATEOAS specifications. Setting the header will result in a FST_ERR_REP_INVALID_PAYLOAD_TYPE error.

This happens because [this code](https://github.com/fastify/fastify/blob/v2.14.1/lib/reply.js#L141-L144) in fastify never runs and never serializes the payload as string.

## To Reproduce

- In a node application, set `request.response.header('Content-Type', 'application/hal+json; charset=utf-8');` or any other custom json format
- Run the application
- The following error occurs:

```bash
FastifyError [FST_ERR_REP_INVALID_PAYLOAD_TYPE]: FST_ERR_REP_INVALID_PAYLOAD_TYPE: Attempted to send payload of invalid type 'object'. Expected a string or Buffer.
    at onSendEnd (/home/user/Projects/nestjs-fastify-sample/node_modules/fastify/lib/reply.js:363:11)
    at onSendHook (/home/user/Projects/nestjs-fastify-sample/node_modules/fastify/lib/reply.js:325:5)
    at _Reply.Reply.send (/home/user/Projects/nestjs-fastify-sample/node_modules/fastify/lib/reply.js:151:3)
    at FastifyAdapter.reply (/home/user/Projects/nestjs-fastify-sample/node_modules/@nestjs/platform-fastify/adapters/fastify-adapter.js:37:25)
    at RouterResponseController.apply (/home/user/Projects/nestjs-fastify-sample/node_modules/@nestjs/core/router/router-response-controller.js:10:36)
    at /home/user/Projects/nestjs-fastify-sample/node_modules/@nestjs/core/router/router-execution-context.js:163:48
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async /home/user/Projects/nestjs-fastify-sample/node_modules/@nestjs/core/router/router-execution-context.js:46:13
    at async Object.<anonymous> (/home/user/Projects/nestjs-fastify-sample/node_modules/@nestjs/core/router/router-proxy.js:8:17)
```

## Expected behavior

Since those are JSON formats, they should be accepted as common JSON payloads.

## Your Environment

- *node version*: 12
- *fastify version*: >=2.14.1
- *os*: Mac

**Comment by Ethan-Arrowood** [comment]
You can provide your own serializer if you want to specially handle these formats.

**Comment by mcollina** [comment]
A better support for custom content-types JSON serializers would be really nice. Something that could be specified alongside the schema.

**Comment by Eomm** [comment]
If we expose this factory in some way:

https://github.com/fastify/fastify/blob/09e3ffb0af1666bb681b8644300e7a8aa97ac219/lib/contentTypeParser.js#L205

the user could just write

```
fastify.addContentTypeParser('application/vnd.api+json', fastify.jsonParser)
fastify.addContentTypeParser('application/ls+json', fastify.jsonParser)
...
```

**Comment by saulotoledo** [comment]
I still think it could be easier. Something like @Eomm's suggestion. I've also tried `fastify.addContentTypeParser()` without success in [this sample project](https://github.com/saulotoledo/nestjs-fastify-content-type-issue) (I am new with Fastify, could you guide me if I did something wrong?)

I was, however, thinking on something like https://github.com/fastify/fastify/pull/2309/files since they are all json files.

**Comment by delvedor** [comment]
Closed in #2315.
